provider/provider.go

```diff
@@ -80,6 +81,18 @@ func (p *dockerNativeProvider) Configure(_ context.Context, req *rpc.ConfigureRe
 	for key, val := range config {
 		p.config[key] = val
 	}
+
+	// Configure a Docker daemon client here even though we won't use it. We want use the native
+	// provider's logic to determine the host address and then set the DOCKER_HOST environment
+	// accordingly so the bridged provider picks it up, too.
```

Comment:
I see - we're using `configureDockerClient` to explicitly set the `DOCKER_HOST` env var.

In the case where the docker host is set via provider config (not env var), `DOCKER_HOST` would then be set to that value, and they would be the same, creating a failsafe.

I'm not super excited about "borrowing" logic from the native provider in this way - truly this should be a (solved) upstream problem (or we should implement the docker host config logic on the hybrid level), but it seems safe, and the early configure failure is certainly nice.

Reply:
I expect that as more people run into the issue with the upstream provider, it'll eventually get fixed there.